### PR TITLE
[BEAM-7063] Enable passing in the worker jar for all tests.

### PR DESCRIPTION
**Pull request by lukecwik**
The Dataflow Runner in Python will only add the worker jar if it is a FnAPI test.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by lukecwik** [comment]
Run Python Dataflow ValidatesRunner

**Comment by lukecwik** [comment]
R: @chamikaramj

**Comment by lukecwik** [comment]
CC: @udim @tvalentyn

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Comment by chamikaramj** [comment]
Run Python PostCommit

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 1]
```diff
@@ -414,11 +414,11 @@ def run_pipeline(self, pipeline, options):
         logging.fatal(
             'Typical end users should not use this worker jar feature. '
             'It can only be used when fnapi is enabled.')
-
-      experiments = ["use_staged_dataflow_worker_jar"]
-      if debug_options.experiments is not None:
-        experiments = list(set(experiments + debug_options.experiments))
-      debug_options.experiments = experiments
+      else:
+        experiments = ["use_staged_dataflow_worker_jar"]
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Please use `add_experiment`: https://github.com/apache/beam/blob/98faabff70b58935b7cbc7efeacc8ec1a850479d/sdks/python/apache_beam/options/pipeline_options.py#L660

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Also please add a unit test.

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: See: https://github.com/apache/beam/blob/2581565fe96f2a1b4be01993831898afefb3e4c7/sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py#L402

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Also, you need to rebase to get past https://github.com/apache/beam/pull/8225.

Comment (lukecwik) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Added unit tests also rebased. PTAL.

**Comment by lukecwik** [comment]
Run Python Dataflow ValidatesRunner

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py [hunk 2]
```diff
@@ -415,6 +416,35 @@ def test_streaming_engine_flag_adds_windmill_experiments(self):
     self.assertIn('enable_windmill_service', experiments_for_job)
     self.assertIn('some_other_experiment', experiments_for_job)
 
+  def test_dataflow_worker_jar_flag_non_fnapi_noop(self):
+    remote_runner = DataflowRunner()
+    print self.default_properties
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py]: Py3-incompatible left-over debug output detected :-)

Comment (lukecwik) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py]: woops

Comment (lukecwik) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py]: Thanks, fixed

**Comment by lukecwik** [comment]
Run Python Dataflow ValidatesRunner

**Pull request merged**